src/waitForElements.js

```diff
@@ -0,0 +1,43 @@
+function waitForElements(
+  callback = () => {},
+  {
+    container = document,
+    timeout = 4500,
+    mutationObserverOptions = {subtree: true, childList: true},
+  } = {},
+) {
+  return new Promise((resolve, reject) => {
+    // Disabling eslint prefer-const below: either prefer-const or no-use-before-define triggers.
+    let lastError, observer, timer // eslint-disable-line prefer-const
+    function onDone(error, result) {
+      clearTimeout(timer)
+      observer.disconnect()
+      if (error) {
+        reject(error)
+      } else {
+        resolve(result)
+      }
+    }
+    function onMutation() {
+      try {
+        const result = callback()
+        if (result) {
+          onDone(null, result)
+        }
+        // If `callback` returns falsy value, wait for the next mutation or timeout.
+      } catch (error) {
+        // Save the callback error to reject the promise with it.
+        lastError = error
+        // If `callback` throws an error, wait for the next mutation or timeout.
+      }
+    }
+    function onTimeout() {
+      onDone(lastError || new Error('Timed out in waitForElements.'), null)
+    }
+    timer = setTimeout(onTimeout, timeout)
+    observer = new MutationObserver(onMutation)
+    observer.observe(container, mutationObserverOptions)
+  })
```

Comment:
Could we run `onMutation` above this line? That way if there's no reason to wait for a mutation to proceed we don't have to wait for one.